### PR TITLE
Feature - query download adapters

### PR DESCRIPTION
**Pull request by JoWilfrid**
While downloading queries results, we can only have it in CSV and that's quite sad because it's not enhanceable from outside the gem.

I've done an adapter system that allows applications using `blazer` to define their own adapters to download queries in multiple formats (e.g: json, pdf, etc.).
Adapters classes are explicitly given while configuring Blazer, in `blazer.yml`.

2 "base" adapters are present in the app: `RenderAdapter` and `SendDataAdapter`.
Basically, every adapter that wants to render something inherits from `RenderAdapter` and every one that wants to make user download something inherits from `SendDataAdapter`.

I've also added some tests.

### CODE REVIEW

**Comment by ankane** [comment]
Hey @JoWilfrid, thanks for another PR 👍  I'm personally not a fan of adding more download formats, as pretty much all languages have CSV support.

**Comment by cycomachead** [comment]
> I'm personally not a fan of adding more download formats, as pretty much all languages have CSV support.

I see the appeal in only maintaining one output format (and for keeping the UI simple). What appeals to me about this method is that those of us who want to could write our own adapters, from really simple JSON ones to more complicated and specialized PDF renders. 

Your call, of course, but I would find an API like this useful and would hopefully get a chance to share any adapters I write.

**Comment by JoWilfrid** [comment]
I did this because I needed to have the ability to download queries as JSON files.
Sure, I could have write a couple of ruby lines to convert CSV into JSON but this approach was supposing that me or another tech person was downloading the file to convert it. The fact is I need it simple for non-tech people.

As I think anyone could have this need one day, I opened this PR :)

**Comment by ankane** [comment]
Hey @JoWilfrid, sorry the discussion stalled. I'd still prefer to err on the side of keeping things simpler, so going to close. Thanks again for putting this together 👍
